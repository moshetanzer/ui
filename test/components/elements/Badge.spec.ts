import { describe, it, expect } from 'vitest'
import { UBadge } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Badge', () => {
  it.each([
    ['with label', { props: { label: 'Badge' } }],
    ['with size', { props: { label: 'Badge', size: 'lg' } }],
    ['with color', { props: { label: 'Badge', color: 'red' } }],
    ['with variant', { props: { label: 'Badge', variant: 'outline' } }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof UBadge.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UBadge)
    expect(html).toMatchSnapshot()
  })
})
