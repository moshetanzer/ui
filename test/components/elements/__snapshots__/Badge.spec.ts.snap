// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Badge > renders with color correctly 1`] = `"<span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-red-500 dark:bg-red-400 text-white dark:text-gray-900">Badge</span>"`;

exports[`Badge > renders with label correctly 1`] = `"<span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 bg-primary-500 dark:bg-primary-400 text-white dark:text-gray-900">Badge</span>"`;

exports[`Badge > renders with size correctly 1`] = `"<span class="inline-flex items-center font-medium rounded-md text-sm px-2.5 py-1.5 bg-primary-500 dark:bg-primary-400 text-white dark:text-gray-900">Badge</span>"`;

exports[`Badge > renders with variant correctly 1`] = `"<span class="inline-flex items-center font-medium rounded-md text-xs px-2 py-1 text-primary-500 dark:text-primary-400 ring-1 ring-inset ring-primary-500 dark:ring-primary-400">Badge</span>"`;
